donut/modules/editor/utils/media_wiki_import.py

```diff
@@ -0,0 +1,142 @@
+#!/usr/bin/env python
+
+import argparse
+from donut.pymysql_connection import make_db
+import re
+import xml.etree.ElementTree as ET
+
+
+def import_file(env, filename):
+    # Create reference to DB
+    db = make_db(env)
+    try:
+        db.begin()
+        tree = ET.parse(filename)
+        root = tree.getroot()
+        query = "INSERT INTO webpage_files(title, last_edit_time, content) VALUES( %s, %s, %s)"
+
+        with db.cursor() as cursor:
+            for elem in root:
+                text = ""
+                date = ""
+                title = ""
+                for ele in elem:
+                    if 'title' in ele.tag:
+                        if 'Image:' in ele.text:
+                            break
+                        title = ele.text
+                    if 'revision' in ele.tag:
+                        for i in ele:
+                            if 'timestamp' in i.tag:
+                                date = i.text
+                            if 'text' in i.tag:
+                                text = i.text
+                if title != '' and text != '' and text is not None:
+                    if date != "":
+                        date = date.replace("T", " ")
+                        date = date.replace("Z", "")
+                    title = title.replace(" ", "_")
+                    text = format_text(text)
+                    cursor.execute(query, [title, date, text])
+        db.commit()
+    finally:
+        db.close()
+
+
+def format_text(text):
+    # Headings
+    matches = re.findall(r"\=+[\+0-9a-zA-Z\(\)\'\* \-\,\?\.]+\=+", text)
+    for i in matches:
+        text = text.replace(i, i.replace("=", "#"))
```

Comment:
Shouldn't the `#` characters only go at the front?

Reply:
Also would be better to use `re.sub()` to avoid searching through `text` every time `replace()` is called

Reply:
Also this regular expression assumes that `[\+0-9a-zA-Z\(\)\'\* \-\,\?\.]` are the only possible characters in links. Can you use `[^=\n]` instead to match all characters until the next `=` or newline?

Reply:
> Shouldn't the `#` characters only go at the front?

1.8.X of showdown updated it to either I guess. I think it's clearer to have it around the heading though.